### ISSUE TITLE
Missing 'state' parameter when using OpenID Connect with Authelia

### ISSUE THREAD

**Issue by jim-kirisame** [pennersr/django-allauth#3631]
When i use allauth with authelia by openid connect, it returns error:

> The state is missing or does not have enough characters and is therefore considered too weak. Request parameter 'state' must be at least be 8 characters long to ensure sufficient entropy.

Authelia enforce oidc client provide state parameter but allauth does not provide when sending auth request.

**Comment by noantiq** [comment]
Hi, since authelia works for me when using oidc (in Paperless-ngx), what is the version and general setup of Authelia you are using and what is the version of allauth the app is using?

**Comment by jim-kirisame** [comment]
> Hi, since authelia works for me when using oidc (in Paperless-ngx), what is the version and general setup of Authelia you are using and what is the version of allauth the app is using?

Hi, the authelia version i'm using is v4.37.5 and the app is [glitchtip](https://gitlab.com/glitchtip/glitchtip-backend) v3.4.6. Sorry i'm not familiar with python so i don't know how to check the allauth version that glitchtip using.

The configuration i'm using is:
> "settings": {
>                        "server_url": "https://auth.example.com/.well-known/openid-configuration"
> }

**Comment by noantiq** [comment]
Me neither unfortunately, but checking their respository I found something that looks promising: https://gitlab.com/glitchtip/glitchtip-backend/-/blob/ef3b24304cb37c1d6ee9ff915005c7c2430090dc/pyproject.toml#L12 for me it looks like they are using almost the latest version. Maybe @pennersr can tell us if the latest version of allauth contains a fix that might explain this, since the version for Authelia worked for me.

@jim-kirisame maybe you can double check that your config looks like the example config that worked for me: https://github.com/pennersr/django-allauth/blob/a0fe9ec0303d6f18e37ac2641589cf94f335c3d4/docs/socialaccount/providers/authelia.rst

**Comment by jim-kirisame** [comment]
@noantiq I have read your document, but glitchip use a web based editor to change these configuration so the config may not exactly the same. I have double checked the client_id, secret and server_url, they are all correct since the login page can successfully redirect to my authelia site and then back to glitchip with a missing state error.

Can you try to login with allauth and check if the request url to authelia contains a 'state' query parameter? There is no such parameter in my login flow so i think that might be a allauth bug.

**Comment by pennersr** [comment]
I've tested this locally -- simply using the example config that @noantiq provided -- and can confirm that allauth is definitely passing along the `state` parameter. Some pointers:

``` python
class OAuth2Adapter(object):
    supports_state = True
```

So, unless a provider overrides the above it does support state. Now, the `class OpenIDConnectOAuth2Adapter(OAuth2Adapter)` class does not override that, meaning, all OpenID Connect providers pass along the state.

Note that this is configuration independent -- so even if your client_id/secret is wrong, it will still pass along state.

The state looks something like this:

```state=DNK5wQm5iNz6McJU```

So its length is 16, which also complies with Authelia's "must be at least be 8 characters long".

All in all, I don't think this ticket is actionable on this end, I will have to close this.

**Comment by pennersr** [comment]
@noantiq Thanks for providing the documentation -- it's merged now.

**Comment by reorx** [comment]
I encountered the same issue when configuring Authelia with GlitchTip, just like @jim-kirisame did.

I found that the OIDC auth request is sent directly via the frontend, so this is definitely not a problem with allauth but rather an issue with GlitchTip handling things incorrectly—not constructing the auth URL through allauth.

Here’s the relevant code: [GlitchTip Frontend OAuth Service](https://gitlab.com/glitchtip/glitchtip-frontend/-/blob/master/src/app/api/oauth/oauth.service.ts#L61).

I will send a bug report to the GlitchTip Frontend repository.
